apps/app_manager.py

```diff
@@ -51,99 +54,115 @@ def load_all_apps(self):
         return base_menu
 
+    def bind_callback(self, app, app_path, menu_name, ordering, subdir_path):
+        if hasattr(app, "callback") and callable(app.callback):  # for function based apps
+            subdir_menu = self.subdir_menus[subdir_path]
+            subdir_menu_contents = self.insert_by_ordering([menu_name, app.callback], os.path.split(app_path)[1],
+                                                           subdir_menu.contents, ordering)
+            subdir_menu.set_contents(subdir_menu_contents)
+            return True
+        if hasattr(app, "on_start") and callable(app.on_start):  # for class based apps
+            subdir_menu = self.subdir_menus[subdir_path]
+            subdir_menu_contents = self.insert_by_ordering([menu_name, app.on_start], os.path.split(app_path)[1],
+                                                           subdir_menu.contents, ordering)
+            subdir_menu.set_contents(subdir_menu_contents)
+        else:
+            print("App \"{}\" has no callback; loading silently".format(menu_name))
+
     def load_app(self, app_path):
         app_import_path = app_path.replace('/', '.')
-        #If user runs in single-app mode and by accident 
-        #autocompletes the app name too far, it shouldn't fail
-        main_py_string = ".main.py" 
+        # If user runs in single-app mode and by accident
+        # autocompletes the app name too far, it shouldn't fail
+        main_py_string = ".main.py"
         if app_import_path.endswith(main_py_string):
             app_import_path = app_import_path[:-len(main_py_string)]
-        app = importlib.import_module(app_import_path+'.main', package='apps')
-        app.init_app(self.i, self.o)
+        app = importlib.import_module(app_import_path + '.main', package='apps')
+        if is_class_based_module(app):
+            zero_app_subclass = get_zeroapp_class_in_module(app)
+            app = zero_app_subclass(self.i, self.o)
+        else:
+            app.init_app(self.i, self.o)
         return app
 
     def get_subdir_menu_name(self, subdir_path):
         """This function gets a subdirectory path and imports __init__.py from it. It then gets _menu_name attribute from __init__.py and returns it. 
         If failed to either import __init__.py or get the _menu_name attribute, it returns the subdirectory name."""
         subdir_import_path = subdir_path.replace('/', '.')
         try:
-            subdir_object = importlib.import_module(subdir_import_path+'.__init__')
+            subdir_object = importlib.import_module(subdir_import_path + '.__init__')
             return subdir_object._menu_name
         except Exception as e:
             print("Exception while loading __init__.py for subdir {}".format(subdir_path))
             print(e)
             return os.path.split(subdir_path)[1].capitalize()
 
-    def get_ordering(self, path, cache={}):
+    def get_ordering(self, path, cache=None):
```

Comment:
So, here, the `cache={}` is actually used as a feature - even though this thing usually trips people up instead of being useful =D If I'm not mistaken, your `if cache is None:` check actually stops the cache from functioning, which is not surprising considering how non-obvious this usage is (and that's certainly my fault). Regardless, I don't remember why I decided to use this bug as a feature when I wrote this - most likely, for fun =) So, feel free to refactor it into an "ordering_cache" attribute of AppManager